[new-parser] NPE caused by an empty DRL input

## Parent issue
- #5678 

## Failing tests
- `org.drools.testcoverage.regression.MultipleKieBaseListenersTest`

## Rule code snippet
```
```

## Error output
```
22:38:57,697 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback-test.xml] at [file:/var/home/jlocker/src/github.com/kiegroup/drools/drools-test-coverage/test-suite/target/test-classes/logback-test.xml]
22:38:57,698 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback-test.xml] occurs multiple times on the classpath.
22:38:57,698 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback-test.xml] occurs at [file:/var/home/jlocker/src/github.com/kiegroup/drools/drools-test-coverage/test-suite/target/test-classes/logback-test.xml]
22:38:57,698 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback-test.xml] occurs at [file:/var/home/jlocker/src/github.com/kiegroup/drools/drools-base/target/test-classes/logback-test.xml]
22:38:57,698 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback-test.xml] occurs at [file:/var/home/jlocker/src/github.com/kiegroup/drools/drools-core/target/test-classes/logback-test.xml]
22:38:57,698 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback-test.xml] occurs at [file:/var/home/jlocker/src/github.com/kiegroup/drools/drools-legacy-test-util/target/test-classes/logback-test.xml]
22:38:57,781 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
22:38:57,781 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
22:38:57,789 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [consoleAppender]
22:38:57,796 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
22:38:57,884 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.kie] to WARN
22:38:57,884 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.drools] to WARN
22:38:57,884 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.drools.compiler.kie.builder.impl] to ERROR
22:38:57,884 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to WARN
22:38:57,884 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [consoleAppender] to Logger[ROOT]
22:38:57,885 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
22:38:57,885 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@1f1c7bf6 - Registering current configuration as safe fallback point

22:38:58.282 [main] WARN  o.k.m.i.embedder.MavenSettings.initSettings:107 - Environment variable M2_HOME is not set
### parse : ANTLR4_PARSER_ENABLED = true
22:38:59.745 [main] ERROR o.d.d.parser.antlr4.DRLParserWrapper.parse:77 - Exception while creating PackageDescr
java.lang.NullPointerException: Cannot invoke "org.antlr.v4.runtime.Token.getStopIndex()" because the return value of "org.antlr.v4.runtime.ParserRuleContext.getStop()" is null
	at org.drools.drl.parser.antlr4.DescrHelper.populateCommonProperties(DescrHelper.java:55)
	at org.drools.drl.parser.antlr4.BaseDescrFactory$Builder.withParserRuleContext(BaseDescrFactory.java:52)
	at org.drools.drl.parser.antlr4.DRLVisitorImpl.visitCompilationUnit(DRLVisitorImpl.java:97)
	at org.drools.drl.parser.antlr4.DRLVisitorImpl.visitCompilationUnit(DRLVisitorImpl.java:83)
	at org.drools.drl.parser.antlr4.DRLParser$CompilationUnitContext.accept(DRLParser.java:435)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visit(AbstractParseTreeVisitor.java:18)
	at org.drools.drl.parser.antlr4.DRLParserHelper.compilationUnitContext2PackageDescr(DRLParserHelper.java:91)
	at org.drools.drl.parser.antlr4.DRLParserWrapper.parse(DRLParserWrapper.java:75)
	at org.drools.drl.parser.antlr4.DRLParserWrapper.parse(DRLParserWrapper.java:55)
	at org.drools.drl.parser.DrlParser.lambda$parse$2(DrlParser.java:191)
	at org.drools.drl.parser.DrlParser.compileWithAntlr4Parser(DrlParser.java:204)
	at org.drools.drl.parser.DrlParser.parse(DrlParser.java:191)
	at org.drools.drl.parser.DrlParser.parse(DrlParser.java:179)
	at org.drools.drl.parser.DrlParser.parse(DrlParser.java:168)
	at org.drools.compiler.builder.impl.resources.DrlResourceHandler.process(DrlResourceHandler.java:55)
	at org.drools.compiler.builder.impl.PackageDescrBuilder.buildResource(PackageDescrBuilder.java:73)
	at org.drools.compiler.builder.impl.PackageDescrBuilder.build(PackageDescrBuilder.java:51)
	at org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.buildPackageDescr(CompositeKnowledgeBuilderImpl.java:173)
	at org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:116)
	at org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:109)
	at org.drools.compiler.kie.builder.impl.AbstractKieProject.buildKnowledgePackages(AbstractKieProject.java:274)
	at org.drools.compiler.kie.builder.impl.AbstractKieProject.buildKnowledgePackages(AbstractKieProject.java:220)
	at org.drools.compiler.kie.builder.impl.AbstractKieProject.verify(AbstractKieProject.java:84)
	at org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildKieProject(KieBuilderImpl.java:285)
	at org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildAll(KieBuilderImpl.java:251)
	at org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildAll(KieBuilderImpl.java:208)
	at org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:149)
	at org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:134)
	at org.drools.testcoverage.common.util.KieUtil.buildAndInstallKieModuleIntoRepo(KieUtil.java:78)
	at org.drools.testcoverage.common.util.KieUtil.buildAndInstallKieModuleIntoRepo(KieUtil.java:72)
	at org.drools.testcoverage.common.util.KieUtil.getKieModuleFromResources(KieUtil.java:249)
	at org.drools.testcoverage.common.util.KieUtil.getKieModuleFromResources(KieUtil.java:238)
	at org.drools.testcoverage.common.util.KieBaseUtil.getKieBaseFromKieModuleFromResources(KieBaseUtil.java:102)
	at org.drools.testcoverage.common.util.KieBaseUtil.getKieBaseFromKieModuleFromDrl(KieBaseUtil.java:115)
	at org.drools.testcoverage.regression.MultipleKieBaseListenersTest.testKnowledgeBaseEventSupportLeak(MultipleKieBaseListenersTest.java:53)
22:38:59.759 [main] ERROR o.d.c.k.b.impl.AbstractKieProject.buildKnowledgePackages:280 - Unable to build KieBaseModel:KieBaseModelName
[0,0]: 
[0,0]: Parser returned a null Package

22:38:59.765 [main] ERROR o.d.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem:165 - KieBuilder errors: [Message [id=1, kieBase=KieBaseModelName, level=ERROR, path=rules1.drl, line=0, column=0
   text=], Message [id=2, kieBase=KieBaseModelName, level=ERROR, path=rules1.drl, line=0, column=0
   text=Parser returned a null Package]]

java.lang.AssertionError: [[Message [id=1, kieBase=KieBaseModelName, level=ERROR, path=rules1.drl, line=0, column=0
   text=], Message [id=2, kieBase=KieBaseModelName, level=ERROR, path=rules1.drl, line=0, column=0
   text=Parser returned a null Package]]] 
Expecting empty but was: [Message [id=1, kieBase=KieBaseModelName, level=ERROR, path=rules1.drl, line=0, column=0
   text=],
    Message [id=2, kieBase=KieBaseModelName, level=ERROR, path=rules1.drl, line=0, column=0
   text=Parser returned a null Package]]

	at org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:169)
	at org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:134)
	at org.drools.testcoverage.common.util.KieUtil.buildAndInstallKieModuleIntoRepo(KieUtil.java:78)
	at org.drools.testcoverage.common.util.KieUtil.buildAndInstallKieModuleIntoRepo(KieUtil.java:72)
	at org.drools.testcoverage.common.util.KieUtil.getKieModuleFromResources(KieUtil.java:249)
	at org.drools.testcoverage.common.util.KieUtil.getKieModuleFromResources(KieUtil.java:238)
	at org.drools.testcoverage.common.util.KieBaseUtil.getKieBaseFromKieModuleFromResources(KieBaseUtil.java:102)
	at org.drools.testcoverage.common.util.KieBaseUtil.getKieBaseFromKieModuleFromDrl(KieBaseUtil.java:115)
	at org.drools.testcoverage.regression.MultipleKieBaseListenersTest.testKnowledgeBaseEventSupportLeak(MultipleKieBaseListenersTest.java:53)
```

--- a/drools-drl/drools-drl-parser-tests/src/test/java/org/drools/drl/parser/antlr4/MiscDRLParserTest.java
+++ b/drools-drl/drools-drl-parser-tests/src/test/java/org/drools/drl/parser/antlr4/MiscDRLParserTest.java
@@ -153,6 +153,13 @@ class MiscDRLParserTest {
     }
 
     @Test
+    void emptySource() {
+        final String source = "";
+        final PackageDescr pkg = parseAndGetPackageDescr(source);
+        assertThat(pkg.getName()).isEmpty();
+    }
+
+    @Test
     void parse_validPackage() {
         final String source = "package foo.bar.baz";
         final PackageDescr pkg = parseAndGetPackageDescr(source);

--- a/drools-drl/drools-drl-parser/src/main/java/org/drools/drl/parser/antlr4/DescrHelper.java
+++ b/drools-drl/drools-drl-parser/src/main/java/org/drools/drl/parser/antlr4/DescrHelper.java
@@ -21,6 +21,7 @@ package org.drools.drl.parser.antlr4;
 import java.util.List;
 
 import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.Token;
 import org.antlr.v4.runtime.tree.TerminalNode;
 import org.drools.drl.ast.descr.AndDescr;
 import org.drools.drl.ast.descr.AttributeDescr;
@@ -38,23 +39,26 @@ public class DescrHelper {
     }
 
     public static <T extends BaseDescr> T populateCommonProperties(T descr, ParserRuleContext ctx) {
+        Token startToken = ctx.getStart(); // Start token is never null.
+        // If the stop token is null, use the start token as both the start end the end.
+        Token stopToken = ctx.getStop() != null ? ctx.getStop() : startToken;
 
         if (descr instanceof ExprConstraintDescr) {
             // Backward Compatibility Notes:
             //   Old DRL6Parser.constraint() has slightly different behavior for ExprConstraintDescr. Keep it for backward compatibility
             //   When we will update LanguageLevel, we can align this with other Descr.
-            descr.setStartCharacter(ctx.getStart().getStartIndex());
-            descr.setEndCharacter(ctx.getStop().getStopIndex());
-            descr.setLocation(ctx.getStart().getLine(), ctx.getStart().getCharPositionInLine());
-            descr.setEndLocation(ctx.getStop().getLine(), ctx.getStop().getCharPositionInLine());
+            descr.setStartCharacter(startToken.getStartIndex());
+            descr.setEndCharacter(stopToken.getStopIndex());
+            descr.setLocation(startToken.getLine(), startToken.getCharPositionInLine());
+            descr.setEndLocation(stopToken.getLine(), stopToken.getCharPositionInLine());
         } else {
-            descr.setStartCharacter(ctx.getStart().getStartIndex());
+            descr.setStartCharacter(startToken.getStartIndex());
             // Backward Compatibility Notes:
             //   Old DRL6Parser adds +1 for EndCharacter (except ExprConstraintDescr). This new parser follows the same to keep the backward compatibility.
             //   However, it doesn't look reasonable. When we will update LanguageLevel, we can remove this +1.
-            descr.setEndCharacter(ctx.getStop().getStopIndex() + 1);
-            descr.setLocation(ctx.getStart().getLine(), ctx.getStart().getCharPositionInLine());
-            descr.setEndLocation(ctx.getStop().getLine(), ctx.getStop().getCharPositionInLine() + ctx.getStop().getText().length() - 1); // last column of the end token
+            descr.setEndCharacter(stopToken.getStopIndex() + 1);
+            descr.setLocation(startToken.getLine(), startToken.getCharPositionInLine());
+            descr.setEndLocation(stopToken.getLine(), stopToken.getCharPositionInLine() + stopToken.getText().length() - 1); // last column of the end token
         }
         return descr;
     }

